stemcell_builder/stages/system_azure_wala/apply.sh

```diff
@@ -46,3 +45,24 @@ cat > $chroot/etc/logrotate.d/waagent <<EOS
     missingok
 }
 EOS
+
+#setup cloud-init
+rm $chroot/etc/cloud/*.cfg
+rm $chroot/etc/cloud/cloud.cfg.d/*.cfg
+cp -f $dir/assets/etc/cloud-init.cfg $chroot/etc/cloud/cloud.cfg
+cp -f $dir/assets/etc/*-*.cfg $chroot/etc/cloud/cloud.cfg.d/
+
```

Comment:
Couple of things, (we started to do them ourselves, but then got a bit confused).

Could we rename `system_azure_wala` to something like `system_azure_init`, since it's doing both the walinux agent and cloud-init now? Makes it less surprising when you find cloud-init things going on in the stage.

It might also help clarity to separate the config files in `assets/etc`, maybe add a `waagent` and `cloud-init` subdirectories.

We noticed that due to the `*-*.cfg` the `90_datasource.cfg` file isn't making it onto the stemcell. We're not sure what this file is doing, but it seemed weird that everything would work without it.

`cloud-init.cfg` also gets copied both into `/etc/cloud` and `/etc/cloud/cloud.cfg.d/`. We're guessing it's only supposed to be in one of the two spots.

Reply:
nice catch about the datasource. I guess it was saved by some smarts // default in cloud-init. This should've explicitly set the datasource for user/metadata to azure (same as bosh agent cloud-init will source the info via different codepaths). setting it explicitly to azure should at least (probably minimally) decrease detection time and make it more reliable.